sparkmagic/sparkmagic/livyclientlib/sparkcommand.py

```diff
@@ -0,0 +1,147 @@
+            return self._pyspark_command(spark_context_variable_name)
+        elif kind == constants.SESSION_KIND_PYSPARK3:
+            return self._pyspark_command(spark_context_variable_name, False)
+        elif kind == constants.SESSION_KIND_SPARK:
+            return self._scala_command(spark_context_variable_name)
+        elif kind == constants.SESSION_KIND_SPARKR:
+            return self._r_command(spark_context_variable_name)
+        else:
+            raise BadUserDataException(u"Kind '{}' is not supported.".format(kind))
+
+    def _pyspark_command(self, spark_context_variable_name, encode_result=True):
+        command = u'{}.toJSON()'.format(spark_context_variable_name)
+        if self.samplemethod == u'sample':
+            command = u'{}.sample(False, {})'.format(command, self.samplefraction)
+        if self.maxrows >= 0:
+            command = u'{}.take({})'.format(command, self.maxrows)
+        else:
+            command = u'{}.collect()'.format(command)
+        # Unicode support has improved in Python 3 so we don't need to encode.
+        if encode_result:
+            print_command = '{}.encode("{}")'.format(constants.LONG_RANDOM_VARIABLE_NAME,
+                                                     conf.pyspark_python_encoding())
+        else:
+            print_command = constants.LONG_RANDOM_VARIABLE_NAME
+        command = u'for {} in {}: print({})'.format(constants.LONG_RANDOM_VARIABLE_NAME,
+                                                    command,
+                                                    print_command)
+        return Command(command)
+
+    def _scala_command(self, spark_context_variable_name):
+        command = u'{}.toJSON'.format(spark_context_variable_name)
+        if self.samplemethod == u'sample':
+            command = u'{}.sample(false, {})'.format(command, self.samplefraction)
+        if self.maxrows >= 0:
+            command = u'{}.take({})'.format(command, self.maxrows)
+        else:
+            command = u'{}.collect'.format(command)
+        return Command(u'{}.foreach(println)'.format(command))
+
+    def _r_command(self, spark_context_variable_name):
+        command = spark_context_variable_name
+        if self.samplemethod == u'sample':
+            command = u'sample({}, FALSE, {})'.format(command,
+                                                      self.samplefraction)
+        if self.maxrows >= 0:
+            command = u'take({},{})'.format(command, self.maxrows)
+        else:
+            command = u'collect({})'.format(command)
+        command = u'jsonlite::toJSON({})'.format(command)
+        command = u'for ({} in ({})) {{cat({})}}'.format(constants.LONG_RANDOM_VARIABLE_NAME,
+                                                         command,
+                                                         constants.LONG_RANDOM_VARIABLE_NAME)
+        return Command(command)
+
+
+    @staticmethod
+    def _records_to_dataframe(records_text, kind):
```

Comment:
It should be easy to pull this method out into `utils.py` without much more refactoring and use it for this class and `SQLQuery`.